xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -796,8 +816,8 @@ bool CPVRChannelGroup::HasNewChannels(void) const
   bool bReturn(false);
   CSingleLock lock(m_critSection);
 
-  for (PVR_CHANNEL_GROUP_MEMBERS::const_iterator it = m_members.begin(); !bReturn && it != m_members.end(); ++it)
-    bReturn = it->second.channel->ChannelID() <= 0;
+  for (std::map<std::pair<int, int>, std::shared_ptr<PVRChannelGroupMember>>::const_iterator it = m_members.begin(); !bReturn && it != m_members.end(); ++it)
+    bReturn = it->second->channel->ChannelID() <= 0;
```

Comment:
get rid of `bReturn`, do early `return true;` on match => range based for loop possible